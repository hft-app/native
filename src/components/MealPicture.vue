<template>
  <div class="fullscreen">
    <div class="photo">
      <img :src="'https://sws2.maxmanager.xyz/assets/' + (meal.photo? meal.photo:
             'fotos/musikhochschule/Speisefotos/0-1/27816947m_dummy_speisen.jpg')"
           :alt="$t('page.meals.imageNotLoaded')">
      <fa-icon icon="times" class="lu-icon lu-times-circle close" @click="$router.back()" />
    </div>
    <div class="data">
      <div class="title">{{ meal.title }}</div>
      <div class="price">{{ meal.price }}</div>
    </div>
  </div>
</template>
<script>
  export default {
    props: {
      meal: {
        type: Object,
        required: true
      }
    },
    created() {
      // go back if the user return from history
      if (!this.meal) this.$router.back();
    }
  }
</script>
<style lang="scss" scoped>
  @import '../colors';

  .fullscreen {
    margin: 0 auto;
    width: 100%;
    max-width: 400px;

    .photo {
      position: relative;

      img {
        width: 100%;
      }

      .close {
        position: absolute;
        right: 0.5em;
        top: 0.5em;
        font-size: 2em;
        background-color: $icon;
        color: #FFF;
        border-radius: 0.5em;
        width: 0.9em;
        height: 0.9em;
        padding: 0.15em;
      }
    }

    .data {
      padding: 2em;

      .title {
        padding-bottom: 2em;
        font-weight: 300;
        font-size: 1.4em;
      }

      .price {
        text-align: right;
        font-weight: 500;
        font-size: 1.5em;
      }
    }
  }
</style>
