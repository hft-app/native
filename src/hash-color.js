import Vue from 'vue';

Vue.directive('hash-color', {
  bind(el, {value}) {
    const colors = ['green', 'blue', 'orange', 'red', 'yellow', 'teal', 'purple', 'pink'];
    let hash = 0;
    for(let i = 0; i < value.length; i++) {
      hash += value.charCodeAt(i);
    }
    hash %= colors.length;
    el.classList.add(colors[hash]);
  }
});
