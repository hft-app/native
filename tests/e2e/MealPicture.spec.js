import {config, shallowMount} from '@vue/test-utils'
import MealPicture from 'components/MealPicture.vue';
import 'hash-color'

config.mocks['$t'] = str => str;

describe('MealPicture', () => {
  test('renders correctly', () => {
    const wrapper = shallowMount(MealPicture, {
      stubs: ['fa-icon'],
      propsData: {
        meal: {
          title: 'Spätzle mit Soß',
          photo: 'photo123.jpg',
          price: '2,85'
        }
      }
    });

    expect(wrapper.element).toMatchSnapshot()
  });

  test('renders correctly - default photo', () => {
    const wrapper = shallowMount(MealPicture, {
      stubs: ['fa-icon'],
      propsData: {
        meal: {
          title: 'Spätzle mit Soß',
          price: '2,85'
        }
      }
    });

    expect(wrapper.element).toMatchSnapshot()
  });
});
