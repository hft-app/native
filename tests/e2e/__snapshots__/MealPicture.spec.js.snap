// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MealPicture renders correctly - default photo 1`] = `
<div
  class="fullscreen"
>
  <div
    class="photo"
  >
    <img
      alt="page.meals.imageNotLoaded"
      src="https://sws2.maxmanager.xyz/assets/fotos/musikhochschule/Speisefotos/0-1/27816947m_dummy_speisen.jpg"
    />
     
    <fa-icon-stub
      class="lu-icon lu-times-circle close"
      icon="times"
    />
  </div>
   
  <div
    class="data"
  >
    <div
      class="title"
    >
      Spätzle mit Soß
    </div>
     
    <div
      class="price"
    >
      2,85
    </div>
  </div>
</div>
`;

exports[`MealPicture renders correctly 1`] = `
<div
  class="fullscreen"
>
  <div
    class="photo"
  >
    <img
      alt="page.meals.imageNotLoaded"
      src="https://sws2.maxmanager.xyz/assets/photo123.jpg"
    />
     
    <fa-icon-stub
      class="lu-icon lu-times-circle close"
      icon="times"
    />
  </div>
   
  <div
    class="data"
  >
    <div
      class="title"
    >
      Spätzle mit Soß
    </div>
     
    <div
      class="price"
    >
      2,85
    </div>
  </div>
</div>
`;
